ais/models.py

```diff
@@ -56,12 +56,34 @@ class StreetAlias(db.Model):
 
     # street_segment = db.relationship('StreetSegment', back_populates='aliases')
 
+class StreetIntersectionQuery(BaseQuery):
+
+    def choose_one(self):
+        #print(self.query)
+        #print(type(self))
+        query = self.order_by(StreetIntersection.geom)
+        return query.limit(1)
+
+    def order_by_intersection(self):
+        return self.order_by(StreetIntersection.geom)
+
 class StreetIntersection(db.Model):
     """An intersection of street centerlines."""
+    query_class = StreetIntersectionQuery
+
     id = db.Column(db.Integer, primary_key=True)
-    street_code_1 = db.Column(db.Text)
-    street_code_2 = db.Column(db.Text)
-    int_ids = db.Column(db.Text)
+    street_1_full = db.Column(db.Text)
+    street_1_name = db.Column(db.Text)
+    street_1_code = db.Column(db.Text)
```

Comment:
index the 2 street code fields (either in the model or in the engine script)
remove index at start of engine script and add back at the end